src/fromager/commands/build.py

```diff
@@ -44,6 +44,12 @@ class BuildSequenceEntry:
     wheel_filename: pathlib.Path
     skipped: bool = False
 
+    @staticmethod
+    def dict_factory(x):
+        return {
+            k: str(v) if isinstance(v, pathlib.Path | Version) else v for (k, v) in x
```

Comment:
I would have expected the 2nd argument to be a tuple, but TIL about union type. Does that come from type hints?

Reply:
That was my first approach - then my editor/pyright complained with the recommendation to use union type not  tuple here. Looks like came in via https://peps.python.org/pep-0604/#isinstance-and-issubclass

Reply:
Nice, thanks for sharing that link! I'm still learning about some of the recent changes to the language after spending a few years away from Python, so every little bit helps. :-)

Reply:
The feature was introduced in Python 3.10 as part of PEP 604, https://docs.python.org/3.10/whatsnew/3.10.html#pep-604-new-type-union-operator